### PR TITLE
[campaign-limit] Adds --campaign-limit to TriggerCampaignCommand

### PR DESCRIPTION
**Pull request by cykonetic**
**Please be sure you are submitting this against the _staging_ branch.**

[//]: # ( Please answer the following questions: )

| Q  | A
| --- | ---
| Bug fix? | No
| New feature? | Revival of an old feature 
| Automated tests included? | No
| Related user documentation PR URL | Help for mautic:campaign:trigger added
| Related developer documentation PR URL | 
| Issues addressed (#s or URLs) | 
| BC breaks? | No
| Deprecations? | No

[//]: # ( Required: )
#### Description: This adds --campaign-limit to the TriggerCampaignCommend
Before 2.12.0 there was a way to limit the number of contacts to process per campaign, so that no one campaign causes a substantial delay to all other campaigns. This PR attempts to bring back this feature.

#### Steps to test this PR:
1. use the --campaign-limit=XXX option with app/console mautic:campaigns:trigger command

### CODE REVIEW

File: app/bundles/CampaignBundle/Entity/CampaignRepository.php [hunk 1]
```diff
@@ -426,15 +430,21 @@ public function getPendingContactIds($campaignId, ContactLimiter $limiter)
         )
             ->setParameter('campaignId', (int) $campaignId);
 
-        $results = $q->execute()->fetchAll();
+        if ($limiter->hasCampaignLimit()) {
+            $q->setMaxResults($limiter->getCampaignLimitRemaining());
```
Comment (heathdutton) [inline app/bundles/CampaignBundle/Entity/CampaignRepository.php]: Doesn't the query already have a max limit from the batch count? If a max limit already exists we'd want to decrease it only if our limit is lower. This should be in updateQueryFromContactLimiter not here.

Comment (cykonetic) [reply app/bundles/CampaignBundle/Entity/CampaignRepository.php]: Good call. Hadn't considered that

**Comment by luizeof** [comment]
can we trigger campaigns by category?

File: app/bundles/CampaignBundle/Entity/CampaignRepository.php (hunk 1 above)
Comment (heathdutton) [reply app/bundles/CampaignBundle/Entity/CampaignRepository.php]: Using values from the limiter outside the limiter to modify queries is contrary to the pattern established.
Please move this logic into updateQueryFromContactLimiter.

Comment (cykonetic) [reply app/bundles/CampaignBundle/Entity/CampaignRepository.php]: I cannot. It is only at the repository level I know if campaign is being considered. Within the limiter, it is only considering the contacts.
Also ContactLimiterTrait uses an instance of ContactLimiter to do the same thing...
`
        if (!$isCount && $limit = $contactLimiter->getBatchLimit()) {
            $qb->setMaxResults($limit);
        }
`

File: app/bundles/CampaignBundle/Entity/CampaignRepository.php [hunk 2]
```diff
@@ -395,6 +395,10 @@ public function getCountsForPendingContacts($campaignId, array $pendingEvents, C
      */
     public function getPendingContactIds($campaignId, ContactLimiter $limiter)
     {
+
```
Comment (escopecz) [inline app/bundles/CampaignBundle/Entity/CampaignRepository.php]: Please remove this empty line. CS fixer is complaining.

**Pull request merged**